web/thesauruses/_meta/control_structures.json

```diff
@@ -10,7 +10,8 @@
       "if_else_conditional",
       "if_elseif_conditional",
       "if_elseif_else_conditional",
-      "ternary_conditional"
+      "ternary_conditional",
+      "defer"
```

Comment:
`defer` isn't idiomatic in most languages, but is a crucial part of Go. That's why I added it; if we don't want to add it for now, let me know.

Reply:
I'm a bit unsure if defer is in the right place here. How i understand them control structures are used to determine which pieces of code are executed.
While defer changes when code is run, apart from terminating the program early it doesn't change if a piece of code is run.
I think something like defer would be better placed somewhere else, for example in functions like [An Introduction to Programming in Go](http://www.golang-book.com/books/intro) and others put it.

Reply:
I'm okay with that too. The official tour of Go _does_ put it under control structures FYI: https://tour.golang.org/flowcontrol/12

I'm not strongly opinionated about this.

Reply:
After further thought, I would have to disagree that `defer` belongs in functions. It is, after all, still a control flow `keyword`. And you may use multiple of them. I think this article explains the role of `defer` pretty well: https://www.digitalocean.com/community/tutorials/understanding-defer-in-go

I realize it's not "traditional" in the sense of whether or not something gets executed.

Reply:
First, I greatly appreciate these discussions! I'm familiar with concepts of Go but haven't really used it, so this is insightful.

Second, I looked it up. It reminds me of an `await` statement in JS. So the idea at least belongs in more than one language and isn't Go-exclusive. (Though upon reading more, stacking defers goes in reverse order, which is interesting.)

I'm rolling this around in my head and think I could see it more in conditionals. While it really only works for calling functions, it's not a part of defining those functions. (Though also in JS, when you call `await` before a function, the function has to also have `async` in front of it, which _would_ be a part of defining the function.) So really it could perhaps belong in both. But for now I think it could belong here.

Since there's no `defer` ID yet, and the language matching code isn't smart enough to combine these together yet when various IDs are missing or there's extras, I wouldn't add it yet. Optionally you can add it and test it to see if it will work.

Reply:
After reading your helpful documentation and how Swift uses `defer` I'm convinced you're right and `defer` does indeed belong in `control_structures.json`.

The category "Conditional Statements (ifs)" still feels off though.

I've looked into it a little and think we might need a category for `try`...`finally` & co. maybe `defer` fit's in there since it's used in a very similar fashion? I've created an issue #194 for the new category. Maybe we can just leave `defer` here until that issue gets resolved?

Reply:
I had pondered if try/catch/finally type things belonged in a Error Handling structure (like in the drop down on the home page). This has me pondering if it belongs here. (I want to say "why not both?" but could mean one could get updated without the other.)

I could see, at least for now, an Error Handling category added and adding several things under there. How's that sound?

Reply:
I am aligned on that, removed 👍